DEVELOPING.md

```diff
@@ -0,0 +1,96 @@
+# Developing Clutch
+
+## Backend
+
+There are many different ways clutch can be configured for backend development.
+Depending on your use case you can enable / disable features in the `clutch-config.yaml`
```

Comment:
do we want to link `clutch-config.yaml` to https://github.com/lyft/clutch/blob/main/backend/clutch-config.yaml, in case users aren't aware this file is provided for them?

Reply:
Actually ill link to our config guide https://clutch.sh/docs/configuration